.github/workflows/misc.yml

```diff
@@ -3,104 +3,33 @@
-      - "gnovm/**"
-      - "tm2/**"
-      - "gno.land/**"
       - "misc/**"
-      - "examples/**"
-      - ".github/workflows/**"
-      # XXX: end copied part, below is the "ideal" paths on which this workflow should run
-      - "misc/genstd/**.go"
-      - "misc/Makefile"
-      - ".github/workflows/misc.yml"
-  push:
-    branches: [ "master" ]
-
-concurrency:
-  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
-  cancel-in-progress: true
+      - ".github/**"
 
 jobs:
-  build:
-    strategy:
-      fail-fast: false
-      matrix:
-        goversion:
-          - "1.22.x"
-        program:
-          - "genstd"
-    runs-on: ubuntu-latest
-    timeout-minutes: 5
-    steps:
-      - name: Install Go
-        uses: actions/setup-go@v5
-        with:
-          go-version: ${{ matrix.goversion }}
-      - name: Checkout code
-        uses: actions/checkout@v4
-
-      - name: go install
-        working-directory: misc
-        run: go install ./${{ matrix.program }}
-
-  test:
+  main:
     strategy:
-      fail-fast: false
-      matrix:
-        goversion:
-          - "1.22.x"
-        args:
-          - _test.genstd
-    runs-on: ubuntu-latest
-    timeout-minutes: 15
-    steps:
-      - name: Checkout code
-        uses: actions/checkout@v4
-
-      - name: Install Go
-        uses: actions/setup-go@v5
-        with:
-          go-version: ${{ matrix.goversion }}
-
-      - name: Set environment variables for debug mode
-        if: env.ACTIONS_STEP_DEBUG == 'true'
-        run: |
-          export LOG_PATH_DIR=${{ runner.temp }}/logs
-          mkdir -p $LOG_PATH_DIR
-
-          echo "LOG_LEVEL=debug" >> $GITHUB_ENV
-          echo "LOG_PATH_DIR=$LOG_PATH_DIR" >> $GITHUB_ENV
-
-      - name: Test
-        working-directory: misc
-        run: |
-          export GOPATH=$HOME/go
-          export GOTEST_FLAGS="-v -p 1 -timeout=30m -coverprofile=coverage.out -covermode=atomic"
-          make ${{ matrix.args }}
-
-      - uses: actions/upload-artifact@v4
-        if: ${{ runner.os == 'Linux' && matrix.goversion == '1.22.x' }}
-        with:
-          name: ${{runner.os}}-coverage-misc-${{matrix.args}}-${{matrix.goversion}}
-          path: ./misc/coverage.out
-
-  upload-coverage:
-    needs: test
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-      - name: Download all previous coverage artifacts
-        uses: actions/download-artifact@v4
-        with:
-          path: ${{ runner.temp }}/coverage
-      - name: Upload combined coverage to Codecov
-        uses: codecov/codecov-action@v4
-        with:
-          directory: ${{ runner.temp }}/coverage
-          token: ${{ secrets.CODECOV_TOKEN }}
-          fail_ci_if_error: ${{ github.repository == 'gnolang/gno' }}
+        fail-fast: false
+        matrix:
+          # fixed list because we have some non go programs on that misc folder
+          program:
+            - autocounterd
+            # - devdeps
```

Comment:
Leftover?

Reply:
Not really, it is go code, but I don't really understand what is doing there. Could you give some insights about why is that needed?

Reply:
![ED4ShZ9UUAAPRQg 612x422](https://github.com/gnolang/gno/assets/16712663/cf91d2a3-0e24-4da9-84c4-4290eec6b1ea)